src/rviz/default_plugin/marker_utils.cpp

```diff
@@ -76,4 +84,346 @@ MarkerBase* createMarker(int marker_type, MarkerDisplay* owner, DisplayContext*
+    checkColor(marker, ss);
+    checkScale(marker, ss);
+    checkPointsEmpty(marker, ss);
+    checkColorsEmpty(marker, ss);
+    checkTextEmpty(marker, ss);
+    break;
+
+  default:
+    ss << "Unknown marker type: " << marker.type ;
+  }
+
+  if(ss.tellp() != 0) //stringstream is not empty
+  {
+    std::string warning = ss.str();
+
+    owner->setMarkerStatus(MarkerID(marker.ns, marker.id), StatusProperty::Warn, warning);
+    ROS_WARN("Marker '%s/%d': %s", marker.ns.c_str(), marker.id, warning.c_str());
+    return false;
+  }
+
+  owner->deleteMarkerStatus(MarkerID(marker.ns, marker.id));
+  return true;
+}
+
+bool checkMarkerArrayMsg(const visualization_msgs::MarkerArray& array, MarkerDisplay* owner)
+{
+  std::vector<MarkerID> marker_ids;
+
+  bool add_marker_in_array = false;
+
+  for(int i = 0; i<array.markers.size(); i++)
+  {
+    if(array.markers[i].action == visualization_msgs::Marker::ADD)
+    {
+      add_marker_in_array = true;
+    }
+
+    if(add_marker_in_array && i != 0 && array.markers[i].action == visualization_msgs::Marker::DELETEALL)
+    {
+      std::stringstream warning;
+      warning << "found a DELETEALL at index " << i << ", previous markers in the MarkerArray will never show";
+      ROS_WARN("MarkerArray: %s", warning.str().c_str());
+      owner->setStatusStd(StatusProperty::Warn, "marker_array", warning.str());
+      return false;
+    }
+    MarkerID current_id(array.markers[i].ns, array.markers[i].id);
+    std::vector<MarkerID>::iterator search = std::lower_bound(marker_ids.begin(), marker_ids.end(), current_id);
+    if (search != marker_ids.end())
+    {
+      std::stringstream ss;
+      ss << "found '" <<  array.markers[i].ns.c_str() << "/" << array.markers[i].id << "' multiple times";
+      ROS_WARN("MarkerArray: %s", ss.str().c_str());
+      owner->setStatusStd(StatusProperty::Warn, "marker_array", ss.str());
+      return false;
+    }
+    else
+    {
+      marker_ids.insert(search, current_id);
+    }
+  }
+  owner->setStatusStd(StatusProperty::Ok, "marker_array", "OK");
+  return true;
+}
+
+
+void checkQuaternion(const visualization_msgs::Marker& marker, std::stringstream& ss)
+{
+  if (marker.pose.orientation.x == 0.0
+    && marker.pose.orientation.y == 0.0
+    && marker.pose.orientation.z == 0.0
+    && marker.pose.orientation.w == 0.0)
+  {
+    addCommaIfRequired(ss);
+    ss << "uninitialized quaternion assuming identity";
+  }
+  else if(!validateQuaternions(marker.pose))
+  {
+    addCommaIfRequired(ss);
+    ss << "unnormalized quaternion in marker message";
+  }
+}
+
+void checkScale(const visualization_msgs::Marker& marker, std::stringstream& ss)
+{
+  // for ARROW markers, scale.z is the optional head length
+  if(marker.type == visualization_msgs::Marker::ARROW && marker.scale.x != 0.0 && marker.scale.y != 0.0)
+    return;
+
+  if(marker.scale.x == 0.0  || marker.scale.y == 0.0  || marker.scale.z == 0.0)
+  {
+    if(marker.type == visualization_msgs::Marker::TRIANGLE_LIST)
```

Comment:
Why mentioning `TRIANGLE_LIST` here separately?

Reply:
It seems not obvious that triangle lists should be scaled in my opinion. The sizes of the triangles should be already specified by the positions of the vertices. While the scaling is a nice convenience feature, forgetting to set the scale might lead to confusion.

The other markers are checked for their specific scale semantics as well.

What do you suggest? Comment the code about the reason for this check? Or a different option?

Reply:
The only thing this `if` does is to add `, TRIANGLE_LIST coordinates are scaled` to the notification.
I believe @rhaschke wondered why that should be necessary.
Given your explanation, I would propose to drop the special case.
As a developer, you would still fill in the missing scale values if you see a warning `scale contains 0.0 in x, y or z` and don't see your mesh, wouldn't you?

Reply:
I agree. I dropped the special case in cf0d693.